user-service/src/main/java/uk/gov/gchq/palisade/service/user/stream/config/AkkaRunnableGraph.java

```diff
@@ -0,0 +1,104 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.gchq.palisade.service.user.stream.config;
+
+import akka.Done;
+import akka.japi.Pair;
+import akka.kafka.ConsumerMessage.Committable;
+import akka.kafka.ConsumerMessage.CommittableMessage;
+import akka.kafka.ProducerMessage;
+import akka.kafka.ProducerMessage.Envelope;
+import akka.kafka.javadsl.Consumer;
+import akka.kafka.javadsl.Consumer.Control;
+import akka.kafka.javadsl.Consumer.DrainingControl;
+import akka.stream.ActorAttributes;
+import akka.stream.Supervision;
+import akka.stream.Supervision.Directive;
+import akka.stream.javadsl.RunnableGraph;
+import akka.stream.javadsl.Sink;
+import akka.stream.javadsl.Source;
+import org.apache.kafka.clients.consumer.ConsumerRecord;
+import org.apache.kafka.clients.producer.ProducerRecord;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import scala.Function1;
+
+import uk.gov.gchq.palisade.User;
+import uk.gov.gchq.palisade.UserId;
+import uk.gov.gchq.palisade.service.user.model.UserRequest;
+import uk.gov.gchq.palisade.service.user.model.UserResponse;
+import uk.gov.gchq.palisade.service.user.model.UserResponse.Builder;
+import uk.gov.gchq.palisade.service.user.service.UserService;
+import uk.gov.gchq.palisade.service.user.stream.ProducerTopicConfiguration;
+import uk.gov.gchq.palisade.service.user.stream.ProducerTopicConfiguration.Topic;
+
+import java.util.Optional;
+import java.util.concurrent.CompletionStage;
+
+/**
+ * Configuration for the Akka Runnable Graph used by the {@link uk.gov.gchq.palisade.service.user.UserApplication}
+ * Configures the connection between Kafka, Akka and the service
+ */
+@Configuration
+public class AkkaRunnableGraph {
+
+    @Bean
+    Function1<Throwable, Directive> supervisor() {
+        return ex -> Supervision.resumingDecider().apply(ex);
+    }
+
+    @Bean
+    RunnableGraph<DrainingControl<Done>> runner(
+            final Source<CommittableMessage<String, UserRequest>, Control> source,
+            final Sink<Envelope<String, UserResponse, Committable>, CompletionStage<Done>> sink,
+            final Function1<Throwable, Directive> supervisionStrategy,
+            final ProducerTopicConfiguration topicConfiguration,
+            @Qualifier("userService") final UserService service) {
+        // Get output topic from config
+        Topic outputTopic = topicConfiguration.getTopics().get("output-topic");
+
+        // Read messages from the stream source
+        return source
+                // Get the user from the userId, keeping track of original message and token
+                .map((CommittableMessage<String, UserRequest> message) -> {
```

Comment:
You are requesting something from an external source, cache now but later maybe LDAP or something really slow. Must use .mapAsync() as per original example code.

Reply:
Updated to a mapAsync() to get the User and then create the UserResponse object.